modules/instrument_builder/js/instrument_builder.instrument.js

```diff
@@ -120,11 +120,20 @@ var Instrument = {
 
                         // Add dropdown and special naming when no date format is set
                         // (i.e when addDateElement() is used)
-                        if (element.Options.dateFormat === "") {
+                        if (!element.Options.dateFormat) {
+                          if (elName === '' || !elName.includes('_date')) {
                             elName = elName + "_date";
-                            dropdown = "select{@}" + elName + "_status" +
+                          }
+                          console.log(element.Description);
+                          if (!element.Description) {
+                            element.Description = '';
+                          }
+                          dropdown = "select{@}" + elName + "_status" +
                             "{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'\n";
                         }
+                        if (!element.Options.dateFormat) {
+                          element.Options.dateFormat = 'Date';
+                        }
```

Comment:
you could also probably move this logic over to the if statement above where you check for !element.Options.dateFormat already

Reply:
@zaliqarosli nice catch! I'm unsure why I did that.